utils/settings.js

```diff
@@ -0,0 +1,42 @@
+const fs = require('fs');
+const homedir = require('os').homedir();
+const path = require('path');
+
+
+// Persistent shell settings
+
+const getShellSettings = () => {
+    const nearPath = path.join(homedir, '.near-config');
```

Comment:
should this be .near or .near-shell or .neardev?

Reply:
also let's make this a constant? same for settings.js

Reply:
> should this be .near or .near-shell or .neardev?

@ilblackdragon I recommended Jane put it here as it fits with the structure I laid out in the NEP.
https://github.com/nearprotocol/NEPs/blob/974887126800be47a58e968fac5b997db7d50f7d/text/0000-near-shell.md#settings-config-and-key-management

```
─ awesome-near-project  ⟵ NEAR dApp
 ├── .near-config       ⟵ Stores shell-experience settings and connection configuration
 │  ├── connections     ⟵ Contains files used to connect and deploy a contract (except keys)
 │  │  ├── default.js   ⟵ Default configuration is for development
 │  │  └── localnet.js  ⟵ Custom connection added by user for localnet
 │  └── settings.js     ⟵ Stores near-shell settings, how shell behaves when run in this project
 └── .near-credentials  ⟵ Previously "neardev" this directory contains private key inforamtion
    └── default         
       └── alice.json   ⟵ NEAR account "alice" has private key information here
```
(the same structure also appears in the home directory)
Based on the [comment](https://github.com/nearprotocol/NEPs/pull/31#issuecomment-606615736) in the shell NEP, I understood that we were not going to put anything in `~/.near` as that's reserved for validator data. We can modify the NEP to be `.neardev` if we want instead of `.near-config`. Currently `neardev` isn't a great name and holds private keys, which the NEP calls to change to a more informative directory `.near-credentials` that folks aren't likely to accidentally revision. Anyway, that's the thought process. Again, we can modify the NEP.

Reply:
moved to consts

Reply:
Do we need configs inside project directory?
I really think we should not store any credentials inside project directory - because that's the easiest way to lose them. E.g. #198 
I also think settings should live in common directory for all projects.

Reply:
@ilblackdragon we don't need credentials, but per-project config is necessary.  At the very least, it's important for making sure that people have updated deps e.g. shell itself. I agree that creds and global config should be in a common dir for everything.

Reply:
@potatodepaulo read the NEP and left comments there - https://github.com/nearprotocol/NEPs/pull/31#pullrequestreview-399439203
High level, I don't think there is need in local configuration folder, especially one that starts with `.` as those directories should be gitignored. `package.json` can contain settings that should be "clonnable".

Reply:
Can we limit the discussion to user defined settings here in order to get the PR in for RL1?